### PR TITLE
[cluster] default non-root

### LINKED ISSUE #6682: [cluster] CVE-2019-5736 fix
This would also have prevented https://github.com/hail-is/hail/issues/6679

Fix is:

```yaml
apiVersion: policy/v1beta1
kind: PodSecurityPolicy
metadata:
  name: non-root
spec:
  privileged: false
  allowPrivilegeEscalation: false
  runAsUser:
    # Require the container to run without root privileges.
    rule: 'MustRunAsNonRoot'
```

### LINKED ISSUE #6693: No container should run as root
Containers running as root are inherently insecure given my current understanding of docker's isolation mechanisms. See: https://github.com/hail-is/hail/issues/6682 for more information on mitigating this issue.

### PR DESCRIPTION
**Pull request by akotlar**
Resolves #6682, resolves #6693

This follows the recommendation from https://kubernetes.io/blog/2019/02/11/runc-and-cve-2019-5736/

This will break any services that request root. I will submit a fix for notebook2 to remove --allow-root.

### CODE REVIEW

File: vdc/k8s-config.yaml [hunk 1]
```diff
@@ -38,6 +49,11 @@ rules:
   resources: ["secrets"]
   resourceNames: ["letsencrypt-config"]
   verbs: ["get"]
+- apiGroups: ['policy']
+  resources: ['podsecuritypolicies']
+  verbs:     ['use']
+  resourceNames:
+  - non-root
```
[the remark this answers is not in the document]
Comment (danking) [reply vdc/k8s-config.yaml]: Erm. Let's define a ClusterRole that enables creating pods everywhere. We should ClusterRoleBinding that to developers.

Then we should create a Role for non-root and a RoleBinding binding it to batch in `batch-pods` and notebook and notebook2 in `default` (we should probably segregate them to another namespace).

[the remark this answers is not in the document]
Comment (danking) [reply vdc/k8s-config.yaml]: I'm not sure how to grant a cluster role binding to developers without listing all of us. I don't understand what, if any, GCP concepts map to k8s "groups". Supposedly, k8s uses whatever groups the authentication provider returns, but I don't know how k8s talks to GCP nor do I know what GCP returns wrt groups.

**Comment by akotlar** [comment]
Prefer to close until after ASHG, as I don't have time to work on this, unless it urgent. @danking ok with this?

**Comment by danking** [comment]
sure

**Pull request closed without merging**